src/plugins/stake-plugins/generic/policyAdapters/CoreumStakeKitAdaptor.ts

```diff
@@ -0,0 +1,429 @@
+import {
+  actionEnter,
+  actionExit,
+  actionPending,
+  ActionRequestDto,
+  APIManager,
+  transactionConstruct,
+  TransactionDto,
+  transactionSubmitHash,
+  yieldGetSingleYieldBalances,
+  yieldYieldOpportunity
+} from '@stakekit/api-hooks'
+import { add, div, eq, floor, gt, mul, sub } from 'biggystring'
+import { EdgeCurrencyWallet, EdgeTransaction } from 'edge-core-js/types'
+
+import { ENV } from '../../../../env'
+import { convertNativeToExchange, DECIMAL_PRECISION } from '../../../../util/utils'
+import { AssetId, ChangeQuote, PositionAllocation, QuoteAllocation, StakePosition } from '../../types'
+import { StakePolicyConfig } from '../types'
+import { StakePolicyAdapter } from './types'
+
+export interface CoreumNativeSkateKitAdapterConfig {
+  type: 'coreum-native-stake-kit'
+  integrationId: string
+  preferredValidatorAddress: string
+  preferredValidatorName: string
+}
+if (ENV.STAKEKIT_API_KEY != null) {
+  APIManager.configure({
+    apiKey: ENV.STAKEKIT_API_KEY,
+    baseURL: 'https://api.stakek.it'
+  })
+}
+
+export const makeSkateKitAdapter = (policyConfig: StakePolicyConfig<CoreumNativeSkateKitAdapterConfig>): StakePolicyAdapter => {
+  if (policyConfig.stakeAssets.length > 1) throw new Error(`Staking more than one assets is not supported for CoreumSkateKitAdapter`)
+  if (policyConfig.rewardAssets.length > 1) throw new Error(`Claim of more than one assets is not supported for CoreumSkateKitAdapter`)
+
+  if (policyConfig.stakeAssets[0].currencyCode !== policyConfig.rewardAssets[0].currencyCode)
+    throw new Error(`Stake and claim of different assets is not supported for CoreumSkateKitAdapter`)
+
+  // Metadata constants:
+  const metadataName = 'Coreum Native Staking'
+  const stakeAsset = policyConfig.stakeAssets[0]
+  const metadataPoolAssetName = `${stakeAsset.currencyCode}`
+
+  const { adapterConfig, stakePolicyId, stakeProviderInfo } = policyConfig
+  const { integrationId, preferredValidatorAddress, preferredValidatorName } = adapterConfig
+
+  async function prepareChangeQuote(
+    wallet: EdgeCurrencyWallet,
+    txs: TransactionDto[],
+    edgeTxs: EdgeTransaction[],
+    allocations: QuoteAllocation[]
+  ): Promise<ChangeQuote> {
+    const networkFee = edgeTxs.reduce((prev, curr) => add(prev, curr.networkFee), '0')
+
+    allocations.push({
+      allocationType: 'networkFee',
+      pluginId: policyConfig.parentPluginId,
+      currencyCode: policyConfig.parentCurrencyCode,
+      nativeAmount: networkFee
+    })
+
+    const approve = async (): Promise<void> => {
+      for (let i = 0; i < txs.length; i++) {
+        const { unsignedTransaction } = txs[i]
+        if (unsignedTransaction == null) continue
+
+        const signedTx = await wallet.signMessage(unsignedTransaction)
```

Comment:
`wallet.signMessage` will correctly sign the transaction? I didn't know this would work.